File: Homework3/assignment3.py
# Multiple Linear Regression

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Crime_Data.csv')
X = dataset.iloc[:, :-1].to_numpy()
y = dataset.iloc[:, -1].to_numpy()


from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.1, #Splits the training and testing set with 10% going to the test
                                                    random_state = 0)

from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train) #Trains the model using the training sets

# Predicting the Test set results


from sklearn.preprocessing import StandardScaler
scx = StandardScaler()
scy = StandardScaler()
X_train = scx.fit_transform(X_train) #Scale X
y_train = scy.fit_transform(y_train.reshape(len(y_train), 1)).flatten() #Scale Y



# Backward Elimination
import statsmodels.api as sm
X_train = sm.add_constant(X_train)

#X_opt = X_train[:, [0,1,2,3,4,5,6]]  Highest p-value is X3 with a p-vale of .945

#X_opt = X_train[:, [0,1,2,4,5,6]] Higest p-value is X5 with a p-value of .232

#X_opt = X_train[:, [0,1,2,4,6]] highest p-valye is X6 with a p-value of .100

X_opt = X_train[:, [0,1,2,4]] #All vars under .05 p-value
regressor_opt = sm.OLS(endog = y_train, exog = X_opt).fit()

regressor_opt.summary() 

print(regressor_opt.params) #First value is intercept, rest of values are coeffiecent values
print(regressor_opt.predict([1,500,50,30]))

File: Homework3/assignment3.r
# Multiple Linear Regression

# Importing the dataset
dataset = read.csv('Crime_Data.csv')


# Splitting the dataset into the Training set and Test set
library(caTools)
set.seed(123)
split = sample.split(dataset$Y, SplitRatio = 0.9)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# Fitting Multiple Linear Regression to the Training set
regressor = lm(formula = Y ~ ., 
               data = training_set)

newdata = data.frame('X1' = c(500),'X2' = c(50),'X4'= c(30))

#Building the optimal model using Backward Elimination

#regressor_opt = lm(formula = Y ~ X1 + X2 + X3 + X4 + X5 + X6, data=training_set) X3 is largest P-value of .799

#regressor_opt = lm(formula = Y ~ X1 + X2 + X4 + X5 + X6, data=training_set) X6 is largest P-value of .245

#regressor_opt = lm(formula = Y ~ X1 + X2 + X4 + X5, data=training_set) X5 is largest P-value of .408

regressor_opt = lm(formula = Y ~ X1 + X2 + X4, data=training_set) #All of the remaing independent vars are under a p-value of .05

y_pred = predict(regressor_opt, newdata = test_set)


y_pred1 = predict(regressor_opt, newdata = data.frame('X1' = c(500),'X2' = c(50),'X4'= c(30)))
print(y_pred1)

print(regressor_opt$coefficients)

